### PR TITLE
Add blueprint-compiler

### LINKED ISSUE #16929: Add blueprint-compiler
### Package name

blueprint-compiler

### Brief description of package

Blueprint is a markup language and compiler for GTK 4 user interfaces.

### URL for package's homepage

https://jwestman.pages.gitlab.gnome.org/blueprint-compiler

### Provide a basic test case to validate the package's functionality.

Blueprint compiler already has tests which can be run via the `meson test -C <build_dir>` command during the build process.

For a manual test, we could

- save the following example code (taken from blueprint-compiler's homepage) as `test.blp`

  ```blueprint
  using Gtk 4.0;
  
  template MyAppWindow : ApplicationWindow {
    default-width: 600;
    default-height: 300;
    title: _("Hello, Blueprint!");
  
    [titlebar]
    HeaderBar {}
  
    Label {
      label: bind MyAppWindow.main_text;
    }
  }
  ```

- and then run `blueprint-compiler compile test.blp` command to see that it spits out some valid XML data.

### MINGW environments where you need the package

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANG32
- [X] CLANGARM64

### Are you willing to submit a PR?

Yes, I'll be submitting a PR very soon.

### PR DESCRIPTION
**Pull request by realmazharhussain**
Closes #16929

### CODE REVIEW

File: blueprint-compiler/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Maintainer: Mazhar Hussain <realmazharhussain@gmail.com>
+
+_realname=blueprint-compiler
+pkgbase=mingw-w64-${_realname}
+pkgname=${MINGW_PACKAGE_PREFIX}-${_realname}
+pkgver=0.6.0
+pkgrel=1
+pkgdesc='A markup language for GTK user interfaces'
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:8]:
```suggestion
pkgdesc='A markup language for GTK user interfaces (mingw-w64)' 
```

File: blueprint-compiler/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+url='https://jwestman.pages.gitlab.gnome.org/blueprint-compiler/'
+license=(LGPL)
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:10]:
```suggestion
license=('spdx:LGPL-3.0-or-later')
```

File: blueprint-compiler/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+arch=(any)
+options=(strip staticlibs)
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:12]: Not required. These are default options.

File: blueprint-compiler/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
+depends=(${MINGW_PACKAGE_PREFIX}-python-gobject)
+makedepends=(${MINGW_PACKAGE_PREFIX}-meson)
+source=(https://gitlab.gnome.org/jwestman/blueprint-compiler/-/archive/v$pkgver/blueprint-compiler-v$pkgver.tar.gz)
+sha256sums=('c9e3652b66803c1de6a24b71f4fa4638b32260b4b04b93c2f3d958e1ce2a175e')
+
+build() {
+  cd "$srcdir"
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:20]: `$srcdir` is default starting directory. Not required to `cd` into it.

File: blueprint-compiler/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  MSYS2_ARG_CONV_EXCL="--prefix=" \
+  "${MINGW_PREFIX}"/bin/meson setup \
+    --prefix="${MINGW_PREFIX}" \
+    $_realname-v$pkgver \
+    build-${MINGW_CHOST}
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:26]:
```suggestion
    build-${MSYSTEM}
```

File: blueprint-compiler/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+check() {
+  cd "$srcdir"
+  meson test -C build-${MINGW_CHOST}
+}
+
+package() {
+  cd "$srcdir"
+  meson install -C build-${MINGW_CHOST} --destdir="$pkgdir"
+}
```
Comment (Biswa96) [inline blueprint-compiler/PKGBUILD:39]: Also install the COPYING file.

File: blueprint-compiler/PKGBUILD (hunk 3 above)
Comment (realmazharhussain) [reply blueprint-compiler/PKGBUILD]: https://github.com/msys2/MINGW-packages/pull/16930/commits/e44b1089ddb042e72650bba2e8c60671e7f638fd

File: blueprint-compiler/PKGBUILD (hunk 4 above)
Comment (realmazharhussain) [reply blueprint-compiler/PKGBUILD]: Commit 8469909842d119c41e543dbf121f282aeb4bf2b4

File: blueprint-compiler/PKGBUILD (hunk 6 above)
Comment (realmazharhussain) [reply blueprint-compiler/PKGBUILD:35]: Commit 0383d0e313c82c362dbbdd7ded6e8cc71cd78ecb

**Comment by mmuetzel** [comment]
This looks like a Python package. Should its name be `python-blueprint-compiler`?

**Comment by realmazharhussain** [comment]
That naming style is used for python libraries. This is an executable program and it being written in Python should be treated as an implementation detail that can change at any time.

Also, MSYS seems to follow Arch Linux package naming scheme and Arch Linux has this package in its repositories and it is named blueprint-compiler there as well.

**Comment by Biswa96** [comment]
> This is an executable program and it being written in Python should be treated as an implementation detail that can change at any time.

I agree with that. The ArchLinux package naming rules can be found here https://wiki.archlinux.org/title/Python_package_guidelines. msys2 mingw packages follow those rules also.

Thank you for the contribution ❤️

**Pull request merged**